packages/@apphosting/adapter-angular/src/utils.ts

```diff
@@ -32,21 +40,113 @@ export async function loadConfig(cwd: string) {
+  const { builder } = workspaceProject.targets.get(target)!;
   if (builder !== "@angular-devkit/build-angular:application") {
     throw new Error("Only the Angular application builder is supported.");
   }
+}
 
-  const buildTarget = {
-    project,
-    target,
-    configuration,
+// Populate file or directory paths we need for generating output directory
+export function populateOutputBundleOptions(outputPaths: OutputPaths): OutputPathOptions {
+  const outputBundleDir = resolve(".apphosting");
+
+  const baseDirectory = fileURLToPath(outputPaths["root"]);
+  const browserRelativePath = relative(baseDirectory, fileURLToPath(outputPaths["browser"]));
+  let serverRelativePath = "server";
+  if (outputPaths["server"]) {
+    serverRelativePath = relative(baseDirectory, fileURLToPath(outputPaths["server"]));
+  }
+
+  return {
+    bundleYamlPath: resolve(outputBundleDir, "bundle.yaml"),
+    outputDirectory: outputBundleDir,
+    baseDirectory,
+    outputBaseDirectory: resolve(outputBundleDir, "dist"),
+    serverFilePath: resolve(outputBundleDir, "dist", serverRelativePath, "server.mjs"),
+    browserDirectory: resolve(outputBundleDir, "dist", browserRelativePath),
   };
+}
+
+// Run build command
+export const build = (cwd = process.cwd()) =>
+  new Promise<OutputPathOptions>((resolve, reject) => {
+    // enable JSON build logs for application builder
+    process.env.NG_BUILD_LOGS_JSON = "1";
+    const childProcess = spawn("npm", ["run", "build"], {
+      cwd,
+      shell: true,
+      stdio: ["inherit", "pipe", "pipe"],
+    });
+    let outputPathOptions = {} as OutputPathOptions;
+    let manifest = {} as ValidManifest;
 
-  const options = await architectHost.getOptionsForTarget(buildTarget);
-  if (!options) throw new Error("Not able to find options for build target.");
-  return options;
+    if (childProcess.stdout) {
+      childProcess.stdout.on("data", (data) => {
+        try {
+          if (data.toString().includes("outputPaths")) {
+            const parsedManifest = JSON.parse(data);
+            // validate if the manifest is of the expected form
+            manifest = buildManifestSchema.parse(parsedManifest);
+            if (manifest["errors"].length > 0) {
+              // errors when extracting manifest
+              manifest.errors.forEach((error) => {
+                logger.error(error);
+              });
+            }
+            if (manifest["warnings"].length > 0) {
+              // warnings when extracting manifest
+              manifest.warnings.forEach((warning) => {
+                logger.info(warning);
+              });
+            }
+            outputPathOptions = populateOutputBundleOptions(manifest["outputPaths"]);
+          }
+        } catch (error) {
+          throw new Error("Build manifest is not of expected structure: " + error);
+        }
+      });
+    } else {
+      throw new Error("Unable to locate build manifest with output paths.");
```

Comment:
let's not throw here, ignore since the SSG / other build scripts might be emitting to stdout